Code/Tools/SceneAPI/SceneUI/ManifestMetaInfoHandler.cpp

```diff
@@ -49,6 +57,56 @@ namespace AZ
+                    AZ_Assert(serializeContext, "No serialize context");
+
+                    // Search the Edit Context for an icon.
+                    // It will have been reflected like:
+                    //  ->Attribute(AZ::Edit::Attributes::Icon, "Editor/Icons/MeshCollider.svg")
+                    auto classData = serializeContext->FindClassData(target.RTTI_GetType());
+                    if (classData && classData->m_editData)
+                    {
+                        auto editorElementData = classData->m_editData->FindElementData(AZ::Edit::ClassElements::EditorData);
+                        if (editorElementData)
+                        {
+                            if (auto iconAttribute = editorElementData->FindAttribute(AZ::Edit::Attributes::Icon))
+                            {
+                                if (auto iconAttributeData = azdynamic_cast<const AZ::Edit::AttributeData<const char*>*>(iconAttribute))
+                                {
+                                    AZStd::string iconAttributeValue = iconAttributeData->Get(&sceneNodeGroup);
+                                    if (!iconAttributeValue.empty())
+                                    {
+                                        iconPath = AZStd::move(iconAttributeValue);
+                                        
+                                        // The path is probably going to be relative to a scan directory,
+                                        // especially if this node was defined in a Gem.
+                                        // If a first check doesn't find the file, then see if the path can
+                                        // be resolved via the asset system, and pull an absolute path from there.
+                                        if (!QFile::exists(QString(iconPath.c_str())))
+                                        {
+                                            AZStd::string iconFullPath;
+                                            bool pathFound = false;
+                                            using AssetSysReqBus = AzToolsFramework::AssetSystemRequestBus;
+                                            AssetSysReqBus::BroadcastResult(
+                                                pathFound, &AssetSysReqBus::Events::GetFullSourcePathFromRelativeProductPath,
+                                                iconPath, iconFullPath);
+                                            if (pathFound)
+                                            {
+                                                iconPath = AZStd::move(iconFullPath);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
There is a bus API I think you can use here instead that does roughly the same thing: `EditorRequests::GetComponentTypeEditorIcon`.

Reply:
I might be reading this function wrong "AZStd::string SandboxIntegrationManager::GetComponentIconPath(const AZ::Uuid& componentType, AZ::Crc32 componentIconAttrib, AZ::Component* component)" but it seems like it would with the 3rd argument being "void*" 

If so, then maybe this method can be updated so that non-Component types could be used?

Reply:
Yeah you can pass a `nullptr` for the `component` or you can just use the `GetComponentTypeEditorIcon(const AZ::Uuid& componentType)` version that just takes a TypeId. You're right that the name is mis-leading, it doesn't have to be an `AZ::Component`. It just searches the serialize context for class data for a given TypeId.

Reply:
For the next pull request I'm working on, I need to pull additional information out of the edit context, so I need this stack of info, anyways.

Reply:
Spoke off-line with @AMZN-stankowi , other attributes are going to be pulled from the `EditContext` in a follow-up PR